app/gui/view/welcome-screen/src/side_menu.rs

```diff
@@ -21,14 +26,15 @@ use web_sys::Element;
 pub struct SideMenu {
     logger:             Logger,
     pub root_dom:       Element,
+    open_project:       OpenProject,
     new_project_button: Element,
     projects_list:      Element,
     closures:           Rc<RefCell<Vec<ClickClosure>>>,
 }
 
 impl SideMenu {
     /// Constructor.
-    pub fn new(logger: &Logger) -> Self {
+    pub fn new(logger: &Logger, open_project: OpenProject, create_project: CreateProject) -> Self {
```

Comment:
Try to inverse - use FRP to pass events to Welcome Screen

Reply:
I decided not to do that.

Reply:
What do you mean by "inverse"? FRP should be used to pass events everywhere. If you are passing events not by FRP, its not done correctly. However, as the first point, I do not understand your comment here, so it needs further explanation.

Reply:
Now the code works like this:
1. WelcomeScreen has its own FRP network, but SideMenu and TemplateCards have not
2. WelcomeScreen prepares two FRP endpoints (`frp::Source`) and gives them to the SideMenu & TemplateCards
3. These FRP endpoints are used in event handlers of HTML elements. When the button is clicked - the specific event fires.
4. WelcomeScreen's FRP handles these events (currently they are connected directly to `open_project` and `create_project` Outputs of the WelcomeScreen).

@farmaazon proposed to avoid passing `frp::Source` and instead use dedicated FRP networks in SideMenu and TemplateCards. The Outputs of these networks should be connected to the WelcomeScreen's network, so the event lifetime would look like this:
```
click -> event fired in event handler -> SideMenu's FRP network -> WelcomeScreen's FRP network -> integration
```

Later we made a decision not to do that, because adding two additional FRP networks introduces more boilerplate and does not bring any improvements.